libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/installer/ComponentInstaller.kt

```diff
@@ -32,6 +41,39 @@ import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
 import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
 import com.mapbox.navigation.ui.maps.view.MapboxCameraModeButton
 
+/**
+ * Install component that renders [LocationPuck].
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+fun ComponentInstaller.locationPuck(
+    mapView: MapView,
```

Comment:
We spoke of this some, but is there a way to remove the `MapView` parameter here?

The component installer could be used in android auto if it used something like a `MapComponentInstaller` 🤔

Reply:
We need `MapView` here in order to access LocationComponentPlugin. `mapView.location` required to create `LocationPuckComponent`.

Remember, these installers are meant to work with MapView, not MapSurface.
For installers that work with MapSurface we need to create separate installers in AA that will accept MapSurface as an argument.

Reply:
I mean, why not use `MapPluginProviderDelegate` to get the `LocationComponentPlugin`

Reply:
Use `MapPluginProviderDelegate` instead of `MapView`? That wouldn't work. We wouldn't be able to acquire `MapboxMap`, nor`Context`.

Reply:
Yeah, would also prefer a simple `MapView` over requiring it all to be passed like this

```
fun ComponentInstaller.locationPuck(
    context: Context,
    mapboxMap: MapboxMap,
    plugins: MapPluginProviderDelegate,
    config: LocationPuckComponentConfig.() -> Unit = {}
): Installation {
```

Another approach could be to define a `MapComponentInstaller`

``` kotlin
fun MapComponentInstaller.locationPuck(
    config: LocationPuckComponentConfig.() -> Unit = {}
): Installation {
```

Reply:
> Another approach could be to define a MapComponentInstaller
>
> ```kotlin
> fun MapComponentInstaller.locationPuck(
>     config: LocationPuckComponentConfig.() -> Unit = {}
> ): Installation {
> ```

We could introduce this once we implement dedicated component installers for `MapView` and `SurfaceMap`.

```kotlin
MapboxNavigationApp.installComponents(this) {
    map(mapView) {  // this: MapComponentInstaller
       locationPuck()
    }
    // and for surface map
    map(surfaceMap) { // this: SurfaceMapComponentInstaller
       locationPuck()
    }
}
```